### PR TITLE
in_recomputing flag for BN

### PR DESCRIPTION
**Pull request by ir5**
This PR will prevent from updating batch normalization statistics more than once when recomputation is performed.
Currently, there is no test for checking the statistics. I just verified the result by manually putting `std::cout << var << std::endl;` in the intermediate of the code and seeing the values.

### CODE REVIEW

File: runtime/ops/normalization.cc [hunk 1]
```diff
@@ -127,7 +127,20 @@ std::tuple<chainerx::Array, XCVMOpaque*, chainerx::Array, chainerx::Array, chain
             x.device().GetBatchNormForwardBackward(result.mean, result.var, epsilon, decay, result.sorted_axis);
     const Array& gamma_reshaped = result.gamma;
     const Array& beta_reshaped = result.beta;
+
```
Comment (shinh) [inline runtime/ops/normalization.cc]: I'm not 100% sure, but I think we can copy `mean` and `var` before passing them to `PreprocessBatchNorm`? I.e.,
```
PreprocessBatchNormResult result;
if (in_recomputing) {
  result = PreprocessBatchNorm(x, s, bias, mean.Copy(), var.Copy(), axes);
} else {
  result = PreprocessBatchNorm(x, s, bias, mean, var, axes);
}
```
In this way, we won't need to revert the change since the inplace modification will be done for copied mean/var?

Comment (ir5) [reply runtime/ops/normalization.cc]: It sounds much better. I would try.

Comment (ir5) [reply runtime/ops/normalization.cc]: updated.
